pkg/geo/geogfn/geohash.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geogfn
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/cockroach/pkg/geo/geopb"
+	"github.com/cockroachdb/cockroach/pkg/testutils/lint/passes/fmtsafe/testdata/src/github.com/cockroachdb/errors"
+)
+
+// GeoHash returns the string representing the geohash from the feature a based on
+// the precision provided p. If no precision is given (0 or -1) it is calculated from
+// the bounding box size. For point features, a default precision of 20 is used.
```

Comment:
maybe let's introduce a constant (e.g. `const GeoHashDefaultPrecision = 20`) to pass in here, instead of allowing 0 / -1.

Reply:
Hello @otan. I have a question with this one: Postgis' ST_GeoHash works both with a given precision and without it. If no precision is given, the function tries to infer a precision based on a bounding box on the feature. This is way I am using two builtins at [geo_builtings.go](https://github.com/cockroachdb/cockroach/pull/50611/files/9b1c911cdbcc9afe2e9d948df49236c65ce7653f#diff-3a753a2870745a81f1da7bd4e876cae2R2809 ). One for when the user choses a precision and one when not.

By using a constant we will be forcing to use the full geohash precision when no precision is given, which will change the behaviour from Postgis'.

Could introducing a new constant work (eg `const GeoHashCalculatePrecision = 0`) and use it when the user wants the precision infered?

Reply:
> Postgis' ST_GeoHash works both with a given precision and without it. If no precision is given, the function tries to infer a precision based on a bounding box on the featur

Ah, thanks - misunderstanding.

> Could introducing a new constant work (eg const GeoHashCalculatePrecision = 0) and use it when the user wants the precision infered?

This sounds like a good idea. What does 0 in PostGIS as a precision give (is it valid)? Is it safer to use -1?

Reply:
> This sounds like a good idea. What does 0 in PostGIS as a precision give (is it valid)? Is it safer to use -1?

It is! I am looking at Postgis source code and it just checks for a positive value. If I put a 0 or a negative value, it just starts calculating the precision, just as if no precision was given.

Will work on that. Hope to have all the changes for later today 😄